### PR TITLE
Add parsing to InternalDerivative aggregation

### PR DESCRIPTION
**Pull request by cbuescher**
Related to #23847, this is a slightly more complicated single value aggregation to parse, since it can 
render two distinct formatted "value_as_string" fields which we need to provide back. Also, the "normalized_value"
that we can parse from the REST response needs to be converted back to the original normalization factor used in
the InternalDerivative object for the calculation.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalNumericMetricsAggregation.java [hunk 1]
```diff
@@ -122,4 +135,48 @@ public boolean equals(Object obj) {
...
+
+            @Override
+            public double parseDouble(String value, boolean roundUp, LongSupplier now) {
+                return 0;
+            }
+
+            @Override
+            public BytesRef parseBytesRef(String value) {
+                return null;
+            }
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalNumericMetricsAggregation.java]: I like the impl from the other PR better as it was throwing UOE in all these methods.

File: core/src/main/java/org/elasticsearch/search/aggregations/pipeline/derivative/InternalDerivative.java [hunk 2]
```diff
@@ -80,17 +92,56 @@ public Object getProperty(List<String> path) {
...
         return builder;
     }
 
+    private static final ObjectParser<Map<String, Object>, Void> PARSER = new ObjectParser<>(
+            "internal_derivative", HashMap::new);
+
+    static {
+        declareCommonField(PARSER);
+        PARSER.declareField((map, value) -> map.put(CommonFields.VALUE.getPreferredName(), value),
+                (p, c) -> InternalMax.parseValue(p, Double.NEGATIVE_INFINITY), CommonFields.VALUE,
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/search/aggregations/pipeline/derivative/InternalDerivative.java]: InternalMax is a leftover here, although it does do the right thing. Shall we call p parser and c context?

**Review by javanna — APPROVED** [review]
left a few comments, along the same lines as #23847

**Comment by javanna** [comment]
here too, I think we can remove the WIP label and label 5.4 as well.

File: core/src/main/java/org/elasticsearch/search/aggregations/pipeline/derivative/InternalDerivative.java (hunk 2 above)
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/search/aggregations/pipeline/derivative/InternalDerivative.java]: yes to both

File: core/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalNumericMetricsAggregation.java (hunk 1 above)
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalNumericMetricsAggregation.java]: will update

**Comment by javanna** [comment]
Closing. We've decided to try a different approach, we'll do it in the feature/client_aggs_parsing.

**Pull request closed without merging**